pkg/webui/console/store/reducers/devices.js

```diff
@@ -104,6 +105,23 @@ const devices = (state = defaultState, { type, payload, event }) => {
       }
 
       return mergeDerived(updatedState, id, derived)
+    case RESET_DEV_SUCCESS:
+      const combinedId = getCombinedDeviceId(payload)
+      const device = state.entities[combinedId]
+      const isOTAA = Boolean(device.supports_join)
+
+      const resetDevice = { ...device }
+      if (isOTAA && resetDevice.session) {
+        // Reset session keys and last seen information for joined OTAA end devices.
+        resetDevice.session.keys = undefined
+        return mergeDerived(
+          { ...state, [combinedId]: resetDevice },
+          combinedId,
+          defaultState.derived,
```

Comment:
```suggestion
          defaultState.derived,
```

Haven't tested but from seeing this, this should not work since you're now spreading an empty object over the existing one, which should have no effect. Or am I missing something?

Reply:
```javascript
const defaultState = { … }
const defaultDerived = {
  lastSeen: undefined,
  uplinkFrameCount: undefined,
  downlinkFrameCount: undefined,
}
…
// Within the reducer:
return mergeDerived(
  { ...state, [combinedId]: resetDevice },
  combinedId,
  defaultDerived,
}
```